### PR TITLE
Feat/friends page

### PR DESCRIPTION
**Pull request by Gderibeaux**
Checkmark: ✓

___x_ Wrote Tests ____ Implemented ____ Reviewed

## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [x] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [] Refactor

## Description of Change:

    description: Added functionality that allows friends list to connect to mock server and allows to add new friends to the state
    
## Requested feedback: 
    
    I'd like feedback on... how to make the website not break when add multiple friends and how to post once we have an api

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [x] My code has no unused/commented out code
    [x] My code has no binding.pry's
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: src/ApiCalls.js [hunk 1]
```diff
@@ -10,4 +10,16 @@ const getUser = phone => {
     .catch(err => console.log(err));
 }
 
-export { getUser };
\ No newline at end of file
+const getFriends = (user_id) => {
+    return fetch(`https://bab2f687-e74e-434e-933e-7c7884a0521d.mock.pstmn.io/api/v1/users/${user_id}/friends`)
+      .then(res => {
```
Comment (jfogiato) [inline src/ApiCalls.js:15]: Looks great!!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -24,6 +25,14 @@ function App() {
     .catch(err => console.log(`There has been an error: ${err}`))
   }, [dispatch]);
 
+  useEffect(() => {
+    getFriends(1)
```
Comment (jfogiato) [inline src/Components/App/App.js:29]: I think it's okay to have this hard keyed for now, but we should really have this referencing `state.user.id` in our Redux slice so it's dynamic, even if we're only working with a single user right now.

File: src/Components/App/App.js [hunk 3]
```diff
@@ -24,6 +25,14 @@ function App() {
...
+      .then(data => {
+        console.log('friends', data.data.friends);
+      })
+      .catch(err => console.log(`There has been an error: ${err}`))
+  }, []);
+
```
Comment (jfogiato) [inline src/Components/App/App.js:35]: Actually ignore my last comment - I think this code block should be removed entirely, right? It's not actually doing anything and lives in `App.js`.

@Gderibeaux It seems like maybe you were using this in `App.js` at first to get it working, then moved to the `FriendsPage` component but forgot to delete your code? Can you confirm?

File: src/Components/FriendsPage/FriendsPage.js [hunk 4]
```diff
@@ -1,41 +1,75 @@
...
+  const [search, setSearch] = useState('');
+  const [friends, setFriends] = useState([]);
+  const [filteredFriends, setFilteredFriends] = useState([]);
+  const [newFriendName, setNewFriendName] = useState('');
 
-  const friends = ['Trevor', 'Gus', 'Joe', 'Jill']
-  const [search, setSearch] = useState('Search Friends');
-  const [filteredFriends, setFilteredFriends] = useState(['Trevor', 'Gus', 'Joe', 'Jill']);
+  useEffect(() => {
+    getFriends(1)
```
Comment (jfogiato) [inline src/Components/FriendsPage/FriendsPage.js:12]: Should pull from `state.user.id`

File: src/Components/FriendsPage/FriendsPage.js [hunk 5]
```diff
@@ -1,41 +1,75 @@
...
+      })
+      .catch(err => console.log(`There has been an error: ${err}`))
+  }, []);
 
-  const displayFriends = filteredFriends.map(friend => {
-    return <div className='long-tile'>{friend}</div>
-  })
+  useEffect(() => {
+    filterFriends(search);
+  });
```
Comment (jfogiato) [inline src/Components/FriendsPage/FriendsPage.js:22]: This seems problematic, thought I'm not entirely sure - with no dependencies listed in the `useEffect` dependency array, this hook would run constantly. If I'm following the logic of how you have this component set up, I think what you could do is add `[search]` as a dependency, so each time that the `search` piece of the form state changes, the `useEffect` reruns and generates a new list of filtered friends. 

Let me know your thoughts and if this suggestion makes sense!

File: src/Components/FriendsPage/FriendsPage.js [hunk 6]
```diff
@@ -1,41 +1,75 @@
...
+  const [newFriendName, setNewFriendName] = useState('');
 
-  const friends = ['Trevor', 'Gus', 'Joe', 'Jill']
-  const [search, setSearch] = useState('Search Friends');
-  const [filteredFriends, setFilteredFriends] = useState(['Trevor', 'Gus', 'Joe', 'Jill']);
+  useEffect(() => {
+    getFriends(1)
+      .then(data => {
+        setFriends(data.data.friends);
+        setFilteredFriends(data.data.friends);
```
Comment (jfogiato) [inline src/Components/FriendsPage/FriendsPage.js:15]: This fetch shouldn't be setting state locally - it should be talking with the Redux root slice and assigning state there. We'll need to change this prior to merging - let me know if you want to chat through implementation - happy to help :)

File: src/Components/FriendsPage/FriendsPage.js [hunk 7]
```diff
@@ -1,41 +1,75 @@
...
+  const handleSearch = event => {
+    setSearch(event.target.value);
+  };
+
+  const handleAddName = () => {
+    if (newFriendName !== '') {
+      const newFriend = { user_id: friends.length + 1, user_name: newFriendName };
+      setFriends([...friends, newFriend]);
+      setFilteredFriends([...filteredFriends, newFriend]);
+      setNewFriendName('');
```
Comment (jfogiato) [inline src/Components/FriendsPage/FriendsPage.js:46]: I think this is the piece that you DM'd about on Slack, but just want to confirm that this piece will just be a request to an end point, then with the response, we'll update `friends` in Redux rootslice state

**Review by trevorfitz0 — COMMENTED** [review]
Good job on all this!

**Pull request merged**